storm-client/src/jvm/org/apache/storm/executor/Executor.java

```diff
@@ -270,6 +273,15 @@ public ExecutorShutdown execute() throws Exception {
 
     @Override
     public void accept(Object event) {
+        if (this.needToRefreshCreds) {
+            LOG.info("The credentials are being updated {}.", executorId);
+            Credentials creds = this.workerData.getCredentials();
+            idToTask.stream().map(Task::getTaskObject).filter(taskObject -> taskObject instanceof ICredentialsListener)
+                    .forEach(taskObject -> {
+                        ((ICredentialsListener) taskObject).setCredentials(creds == null ? null : creds.get_creds());
+                    });
+            this.needToRefreshCreds = false;
```

Comment:
I think maybe we should first set  this.needToRefreshCreds to false and then update the creds.  Imagine we uploaded credentials twice in a row and this loaded the first creds and the thread paused.  It would then clear the update flag and miss the second credentials upload.

Reply:
Addressed.